python_modules/dagster/dagster/__init__.py

```diff
@@ -253,19 +249,44 @@
 
 from dagster.config.source import BoolSource, StringSource, IntSource  # isort:skip
 
+# NOTE: Unfortunately we have to declare all dynamic attributes (i.e. lazy loaded and deprecated
+# names) twice-- the top level declaration informs pylint that the module contains the attribute
+# (this prevents generating noisy lint errors for users), but an entry in a richer data structure is
+# required for to provide the information needed to resolve the attribute. Note that the
+# typing.Annotated functionality introduced in Python 3.9 (or available through the
+# `typing-extensions` package) would allow us to avoid the double declaration, e.g.:
+#
+# EventMetadataEntry: Annotated[Type[MetadataEntry], MetadataEntry, "0.15.0"]
+
 # ########################
-# ##### DEPRECATED ALIASES
+# ##### LAZY LOAD
 # ########################
 
-# NOTE: Unfortunately we have to declare deprecated aliases twice-- the top
-# level declaration informs pylint that the module contains the attribute (this
-# prevents generating noisy lint errors for users), but the entry in
-# `_DEPRECATED` is required  for us to generate the deprecation warning. Note
-# that the typing.Annotated functionality introduced in Python 3.9 (or
-# available through the `typing-extensions` package) would allow us to avoid
-# the double declaration, e.g.:
-#
-# EventMetadataEntry: Annotated[Type[MetadataEntry], MetadataEntry, "0.15.0"]
+EventLogEntry: Any
+EventLogRecord: Any
+EventRecordsFilter: Any
+RunShardedEventsCursor: Any
+
```

Comment:
I guess this dimension is quite unfortunate

Reply:
One more potential benefit of bringing in `typing-extensions` as dependency... (see longish comment about dynamic imports)

Reply:
To clarify Im speaking to the type information being set to `Any` for these objects. I don't see how whats discussed in the comment would work for lazy loading. Am I missing something?

Reply:
I think you're right on second thought-- but luckily it's now a moot point bc found a better solution using `typing.TYPE_CHECKING`.